evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -44,24 +66,19 @@ def fit(self, X, y=None):
 
         Returns:
             self
+
         """
         X = infer_feature_types(X)
+
         nan_ratio = X.ww.describe().loc["nan_count"] / X.shape[0]
         self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
 
-        # Not using select because we just need column names, not a new dataframe
-        natural_language_columns = [
-            col
-            for col, ltype in X.ww.logical_types.items()
-            if type(ltype) == NaturalLanguage
-        ]
-        if natural_language_columns:
-            X = X.ww.copy()
-            X.ww.set_types({col: "Categorical" for col in natural_language_columns})
+        X, _ = self._drop_natural_language_columns(X)
+        X = self._set_boolean_columns_to_categorical(X)
 
-        # Convert all bool dtypes to category for fitting
-        if (X.dtypes == bool).all():
-            X = X.astype("category")
+        # If the Dataframe only had one natural language column, do nothing.
```

Comment:
Nit pick, could have been more than one natural language column right? This happens whenever our input data *only* has natural language columns

Reply:
Thank you :)